src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -691,18 +691,30 @@ def auto_transfer(self, service_id: uuid.UUID) -> inmanta_lsm.model.ServiceInsta
             transfer_type=inmanta_lsm.const.TransferTrigger.AUTO,
         )
 
-        def next_state(state: str) -> None:
+        def next_state(state: str, is_error_transition: bool = False) -> None:
             """
             Apply this state to our service, if it is different from the current
             state, also increment the version
 
             :param state: The new state to apply
+            :param is_error_transition: Is it an error transition?
             """
             if service.state == state:
                 return
 
             service.last_updated = datetime.datetime.now()
             service.version += 1
+
+            try:
+                is_preserving_same_desired_state = (
+                    is_error_transition and transfer.error_same_desired_state
+                ) or transfer.target_same_desired_state
```

Comment:
```suggestion
                is_preserving_same_desired_state = transfer.error_same_desired_state if is_error_transition else transfer.target_same_desired_state
```
Isn't this more clear?